### PR TITLE
Preserve

### PR DESCRIPTION
**Pull request by MasumaJaffery**
Hello Code Reviewer!
Hope you are doing great;
In this Milestone:
- Store data in files.
- Run a program using the command line.
Thanks for the time to review our PR!
Happy Reviewing. 🤓 
Stay Blessed. ⭐

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -10,16 +10,16 @@ AllCops:
...
-  Max: 20
+  Max: 120
 Metrics/AbcSize:
   Max: 50
 Metrics/ClassLength:
   Max: 150
 Metrics/BlockLength:
   IgnoredMethods: ['describe']
-  Max: 30
+  Max: 130
```
Comment (danmainah) [inline .rubocop.yml]: - [ ] @Team it's not good practice to change the default settings of your linter configuration, we are required to use them in default mode to help us write clean and quality code.

File: data/data_manager.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  def load_rentals
+    return unless File.exist?('rentals.json')
+
+    json_str = File.read('rentals.json')
+    @rentals = JSON.parse(json_str).map do |rental_data|
+      book = @books.find { |b| b.id == rental_data['book_id'] }
+      person = @people.find { |p| p.id == rental_data['person_id'] }
+      Rental.new(rental_data['date'], book, person)
+    end
+  end
```
Comment (danmainah) [inline data/data_manager.rb]: - [ ] @Team Great work on your app, however, I observed the bug below when checking if the app, ` When a  user deletes the rentals.json file and exits the app, then opens the app and adds new rental and exit the app again, when reopening the app I get the error below`, it seems deletion of rentals.json and adding new rentals causes this issue.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @MasumaJaffery and @najibullahjafari , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Ensured data is saved when the user exits the app on the JSON files.
✔️ On startup of the app, all data is loaded from files, and the app handles cases where one or all json files are missing.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Johnadibe — APPROVED** [review]
# Status: Approved ✔️
### Hi @MasumaJaffery and @najibullahjafari👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

LeonardoInGatsby

## Highlights
- On exit from your app, all the data (books, people, and rentals) are saved into JSON files✔️
- On startup of your app, all data are loaded from files.✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**